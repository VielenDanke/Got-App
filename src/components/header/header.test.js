import React from 'react'
import Header from "./header"
import {shallow} from "enzyme"

describe("Testing <Header/>", () => {
    it('Header have rendered correctly', () => {
        const char = shallow(<Header/>)
        expect(char).toMatchSnapshot()
    })
})