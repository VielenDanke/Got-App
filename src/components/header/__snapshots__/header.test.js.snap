// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Header/> Header have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="header-title"
        >
          <Link
            replace={false}
            to="/"
          >
            Game of Thrones DB
          </Link>
        </h3>,
        <ul
          className="header-list"
        >
          <li>
            <Link
              replace={false}
              to="/characters/"
            >
              Characters
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/houses/"
            >
              Houses
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/books/"
            >
              Books
            </Link>
          </li>
        </ul>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
          >
            Game of Thrones DB
          </Link>,
          "className": "header-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Game of Thrones DB",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Game of Thrones DB",
          "type": [Function],
        },
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="/characters/"
              >
                Characters
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/houses/"
              >
                Houses
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/books/"
              >
                Books
              </Link>
            </li>,
          ],
          "className": "header-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/characters/"
              >
                Characters
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Characters",
                "replace": false,
                "to": "/characters/",
              },
              "ref": null,
              "rendered": "Characters",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/houses/"
              >
                Houses
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Houses",
                "replace": false,
                "to": "/houses/",
              },
              "ref": null,
              "rendered": "Houses",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/books/"
              >
                Books
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Books",
                "replace": false,
                "to": "/books/",
              },
              "ref": null,
              "rendered": "Books",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="header-title"
          >
            <Link
              replace={false}
              to="/"
            >
              Game of Thrones DB
            </Link>
          </h3>,
          <ul
            className="header-list"
          >
            <li>
              <Link
                replace={false}
                to="/characters/"
              >
                Characters
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/houses/"
              >
                Houses
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/books/"
              >
                Books
              </Link>
            </li>
          </ul>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              Game of Thrones DB
            </Link>,
            "className": "header-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Game of Thrones DB",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Game of Thrones DB",
            "type": [Function],
          },
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  replace={false}
                  to="/characters/"
                >
                  Characters
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/houses/"
                >
                  Houses
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/books/"
                >
                  Books
                </Link>
              </li>,
            ],
            "className": "header-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/characters/"
                >
                  Characters
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Characters",
                  "replace": false,
                  "to": "/characters/",
                },
                "ref": null,
                "rendered": "Characters",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/houses/"
                >
                  Houses
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Houses",
                  "replace": false,
                  "to": "/houses/",
                },
                "ref": null,
                "rendered": "Houses",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/books/"
                >
                  Books
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Books",
                  "replace": false,
                  "to": "/books/",
                },
                "ref": null,
                "rendered": "Books",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
